### PR TITLE
Add Decoded Spot Table

### LINKED ISSUE #1076: SpaceTx output format for decoded spots
The `IntensityTable` has great features as an internal `starfish` data structure and the corresponding `.netcdf` file is the natural representation of that on disk.  For users to work on downstream processing such as segmentation based on spots and spatial maps of decoded mRNA counts, we really only need a format that contains the following information for every mRNA spot after decoding:
x,y,z coordinates in physical units
target name
optional: quality score (e.g. from decoding)  

I think a delimited text format would work fine for this- let's keep it simple. 
This could be written to disk directly from the `IntensityTable` but I would also appreciate having access to it as a data structure...

### PR DESCRIPTION
**Pull request by ambrosejcarr**
Adds a `DecodedSpotTable` and some basic serialization tooling to address #1076. Users who want the underlying dataframe can get it with `DecodedSpotTable.data`. 

See documentation included in PR for more details. 

closes #1076

### CODE REVIEW

**Comment by ambrosejcarr** [comment]
> Let's run with this and discuss with WG5 users about how to organize multi-cell assignment with probabilities within this framework.

Sounds good. I'd also be fine adding a sparse matrix and saving both tables in an archive. 

What would be your preferred way to proceed with WG5? CC then on this PR? email them? post this in slack?

File: docs/source/sptx-format/output_formats/DecodedSpots/index.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+associated with the listed target.
+
+:code:`cell (int):` the identifier of the cell that contains this spot.
+
+:code:`cell_probability (float):` the quality of a cell association, or probability that the spot
+is associated with the listed cell.
+
+Implementation
+--------------
+Starfish Implements the :py:class:`DecodedSpots` as a wrapper for the :code:`pd.DataFrame` object
```
Comment (ttung) [inline docs/source/sptx-format/output_formats/DecodedSpots/index.rst]:
```suggestion
Starfish implements the :py:class:`DecodedSpots` as a wrapper for the :code:`pd.DataFrame` object
```

File: starfish/intensity_table/intensity_table.py [hunk 2]
```diff
@@ -406,6 +406,19 @@ def to_features_dataframe(self) -> pd.DataFrame:
         """
         return pd.DataFrame(dict(self[Features.AXIS].coords))
 
+    def to_decoded_spots(self) -> DecodedSpots:
+        """
+        Generates a dataframe containing decoded spot information. Guaranteed to contain physical
+        spot coordinates (z, y, x) and gene target. Does not contain pixel coordinates.
+        """
+        if Features.TARGET not in self.coords.keys():
+            raise RuntimeError(
```
Comment (ttung) [inline starfish/intensity_table/intensity_table.py]:
```suggestion
            raise ValueError(
```

File: starfish/test/test_decoded_spots.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    # mock decoder run by adding target list
+    data[Features.TARGET] = (Features.AXIS, list('abcde'))
+
+    ds = data.to_decoded_spots()
+
+    assert ds.data.shape[0] == 5
+
+    # test serialization
+    tempdir = tempfile.mkdtemp()
+    filename = os.path.join(tempdir, 'spots.csv')
```
Comment (ttung) [inline starfish/test/test_decoded_spots.py]: You should probably just use: `with tempfile.NamedTemporaryFile(...)`

File: starfish/types/_decoded_spots.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+from starfish.types import Coordinates, Features
+from ._validated_table import ValidatedTable
+
+
+class DecodedSpots(ValidatedTable):
+
+    required_fields = {
+        Coordinates.X.value,          # spot x-coordinate
+        Coordinates.Y.value,          # spot y-coordinate
+        Features.TARGET,     # spot gene target
```
Comment (ttung) [inline starfish/types/_decoded_spots.py]: some wonky alignment going on.

**Pull request merged**